### DISCUSSION TITLE
Conditional confirmation step on collection

### DISCUSSION THREAD

**Discussion by jdjoshuadavison** [directus/directus#19662] [Feature Requests]
### Summary

Add option to configure a conditional confirmation step with customisable message on a collection so that admins can set a message to be displayed when a user saves a record and certain conditions are met. 

### Basic Example

If status === "published" require confirmation with message "you're editing a published record, are you sure you want to save?"

### Motivation

Motivation for adding a configurable confirmation step: to require people to think twice before saving their content.
Motivation for making it conditional: because confirmation steps are annoying when its not critical.
Motivation for making it customisable: allows admins to set a specific message to prompt the user to think about why you're asking them to confirm.

### Detailed Design

In the collection setup section add a toggle called "require confirmation" and a "customise" button.

The customise button will open a drawer that has a field to add condition rules (using a similar interface to the rules interface in the conditions section on a field) and a text input field to set a custom message.

### Must Have:
- Option to toggle on or off (toggled off by default)
- Option to set conditions for displaying the confirmation step
- Option to set custom message to be displayed in the confirmation dialogue

### Should Have:
-

### Could Have:
- Option to customise the accept and cancel button text
- Option to require ticking a box with customisable label before being able to click accept
- Ability to create multiple condition sets, each with a custom message

### Won't Have:
- Errors :zany_face:

### Drawbacks

None.
It could be annoying if admins enabled this setting for all collections without any conditions when not absolutely necessary, but that's unlikely and there is no higher risk of misuse/misconfiguration with this feature than any other feature and the negative impact is very very low.

### Alternatives

The impact of not having this is that users will continue to save records without due diligence. Sometimes users are saving to the current record when they intended to save as a copy, sometimes users need that extra prompt to go back and check their spellings before hitting save, sometimes users think they're editing something that's in draft and don't realise they're editing a published record. A confirmation step will help minimise these issues.

I was thinking about a way to do this using flows: add a tickbox to the data model that is required to be checked to be valid if conditions are met, then have a non-blocking flow uncheck it after its been saved. This might work, and if it does it will suffice for now, but a native feature would have less performance impact, be more intuitive and I think it would benefit lots of users.

### Adoption Strategy

Nothing required. The setting will be disabled by default and admins can simply toggle it on and configure it if they want to.

### Unresolved Questions

Is this a good idea or a great idea? 😝

**Comment by jeffsmonteiro** [comment]
Related (complementary) feature request
=> https://github.com/directus/directus/discussions/18634

**Reply by jdjoshuadavison** [reply]
I don’t see anything relating these requests. Did you link the wrong item, perhaps?

**Reply by jeffsmonteiro** [reply]
Let's go! You would like to provide a modal for confirming an action, and you would like to customize button texts and probably define what will happen to the user. If my understanding is correct, doesn't it make sense that the modal with customized messages should not be created for this?

I didn't say the two proposals to solve the same problem, but rather that they are related. In this case, you might think they are complementary. And as the other proposal already has a greater number of upvotes, it might be a good idea for you to leave your considerations there so that you have a visibility hook for your proposal! ;)

**Comment by BradleySchapf** [comment]
Would really love to have this type of confirmation modal available. I don't want people changing sensitive fields without being aware of the consequences.

**Reply by LudovicSterlin** [reply]
Same here, especially when an item is used across numerous other "parent" items and the update trigger hooks async update on the parents/childs items.
